File: modules/retrofit/src/main/java/com/sichard/retrofit/mvp/MvpBaseActivity.java
package com.sichard.retrofit.mvp;

import android.os.Bundle;
import android.support.annotation.Nullable;

import com.android.sichard.common.BaseActivity;

/**
 * <br>类表述：
 * <br>详细描述：
 * <br><b>Author Caoshichao </b></br>
 * <br><b>Date  2020/4/2</b></br>
 */
public abstract class MvpBaseActivity<P extends BasePresenter> extends BaseActivity implements IView{
    public P mPresenter;
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mPresenter = createPresenter();
        if (mPresenter != null) {
            mPresenter.attachView(this);
        }
    }

    protected abstract P createPresenter();

    @Override
    protected void onDestroy() {
        super.onDestroy();
        if (mPresenter != null) {
            mPresenter.detachView();
        }
    }
}

File: modules/retrofit/src/main/java/com/sichard/retrofit/mvp/MvpBaseActivity.kt
package com.sichard.retrofit.mvp

import android.os.Bundle
import com.android.sichard.common.BaseActivity

/**
 * <br></br>类表述：
 * <br></br>详细描述：
 * <br></br>**Author Caoshichao **
 * <br></br>**Date  2020/4/2**
 */
abstract class MvpBaseActivity<P : BasePresenter<*, V>, V : IView> : BaseActivity(), IView {
    lateinit var mPresenter: P
    @Suppress("UNCHECKED_CAST")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mPresenter = createPresenter()
        mPresenter.attachView(this as V)
    }

    protected abstract fun createPresenter(): P
    override fun onDestroy() {
        super.onDestroy()
        mPresenter.detachView()
    }
}